apps/web/src/network/Network.tsx

```diff
@@ -226,6 +276,21 @@ export default function GraphNetwork() {
       _network.stopSimulation();
     });
 
+
+    _network.on('click', (params) => {
+      const { ui, data } = appStore.getState();
+      if (params.nodes.length > 0) {
+        highlightDeepDependencies(data, ui.network.selectedNodeId, false);
+        const nodeId = params.nodes[0]
+        if (ui.network.dependencies.deep.active) {
+          highlightDeepDependencies(data, nodeId, true);
+          ui.network.selectedNodeId = nodeId
```

Comment:
Instead of directly mutating the `ui` state, could you please stick with the current approach which goes from a action dispatching (within a use case/function) and a reducer to mutate the state? That way, we can reason about UI states with only reducers/actions, otherwise we might end up with side-effects that can't be easily tracked.

Moreover that small use case could have its own unit test asserting that the final state (post dispatching) is the one expected given the provided parameters.

Reply:
I'll look into it. Not too familiar with this pattern.
I assume you mean this when you're talking about reducers: https://kentcdodds.com/blog/the-state-reducer-pattern-with-react-hooks

Reply:
Yeah but not using React.useReducer, you can refer to other examples in the code base. Basically all actions that currently update both the ui and the core state follow the same pattern

Reply:
Resolved in https://github.com/antoine-coulon/skott/pull/180/commits/808b98568e127673c1fa51670f7502dda35811b2

Reply:
Sorry, not too sure what that would look like. There are so many wrappers, I'm not sure which functions to write/change

My understanding is that you would like to trigger UI updates through events, which is handled in reducers.

Now how do you trigger events?
Through `callUseCase`?
So I'll have to provide a function that returns a function to dispatch the `select_node` event?

But then I'll have to keep track of previously selected `nodeId` in the state too

Reply:
Since `reducer` is a kind of event listener, maybe I should focus on `appStore.dispatch` to trigger events in the `_network` click event listener.

Definitely need to keep a history of the previously clicked `nodeId` in the state then.

Then UI changes are done in `networkUIReducer`, listening to some `select_node` action

Reply:
Resolved in https://github.com/antoine-coulon/skott/pull/180/commits/e4ad4d54e28d4d6ef0465ca4b4a8a77e1c6257bc